app/script/event/EventService.js

```diff
@@ -134,6 +133,31 @@ z.event.EventService = class EventService {
     return this.storageService.update(this.EVENT_STORE_NAME, event.primary_key, event).then(() => event);
   }
 
+  /**
+   * Delete an event from a conversation. Duplicates are delete as well.
+   *
+   * @param {string} conversationId - ID of conversation to remove message from
+   * @param {string} eventId - ID of the actual message
+   * @returns {Promise} Resolves with the number of deleted records
+   */
+  deleteEvent(conversationId, eventId) {
+    return this.storageService.db[this.EVENT_STORE_NAME]
+      .where('id')
```

Comment:
Minor pref improvements here (selecting first using the event id and then filtering with the right conversation, before it was the other way around).